tests/TestCase/Collection/CollectionTest.php

```diff
@@ -1930,12 +1930,12 @@ public function testZipWith(): void
         $zipped = $collection->zipWith([3, 4], function ($a, $b) {
             return $a * $b;
         });
-        $this->assertEquals([3, 8], $zipped->toList());
+        $this->assertEquals([[3], [8]], $zipped->toList());
```

Comment:
This change will break user land code and how a 'zipWith' operation should work. I'm pretty sure the intent of this method is to be an analog to [haskell's zipWith](http://zvon.org/other/haskell/Outputprelude/zipWith_f.html)

Reply:
right, but how to we create that with MultipleIterator::current?

Reply:
Maybe we shouldn't be extending `MultipleIterator` if it requires us to return an array?

Reply:
I guess we can just leave the ReturnTypeWillChange attribute for 5.0 and hope nothing else affects it.

Reply:
I think we should be setting the php compatibility to `>8,<8.2` based on how the recent PHP releases have been going 😛